src/main/webapp/user.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Check if the user is logged in.
+ */
+function authenticateUser() {
+  const currentWindowLocation = window.location.href;
+  const currentUrl = new URL(currentWindowLocation);
+  return queryAuthenticationServer(currentUrl).then((userAuthenticationStatus) => {
+    return userAuthenticationStatus;
+  });
+}
+
+/**
+ * Send GET request to UserAuthenticationServlet with the current page url
+ * as the query parameters.
+ */
+function queryAuthenticationServer(currentUrl) {
+  const requestUrl = new URL('/user-authentication', currentUrl);
+  const queryParams = new QueryParams(currentUrl);
+  requestUrl.search = new URLSearchParams(queryParams).toString();
+
+  return fetch(requestUrl, {method: 'GET'})
+      .catch(error => console.log(error))
+      .then((response) => {
+        return response.json();
+      });
+}
+
+class QueryParams {
+  constructor(currentUrl) {
+    this.currentUrl = currentUrl;
+  }
+}
+
+function addLinktoLogoutButton(logoutUrl) {
+  const logoutButton = document.getElementById('logout');
+  logoutButton.addEventListener('click', () => {
+    window.location.replace(logoutUrl);
+  });
+}
+
+/**
+ * Saves the currently logged in user in the database.
+ */
+function saveUserInDatabase() {
+  return fetch('/new-user-servlet', {method: 'POST'})
+      .catch(error => "saveUserInDatabase: failed to post new user: " + error);
+}
+
+/**
+ * Checks if the currently logged in user is already saved in the database.
+ */
+function userPresentInDatabase() {
```

Comment:
Do we really need to expose this to client? Would it be better if this check as well as the save-user are done silently by the servlet?

Reply:
Moved check to the servlet.